iac/mainnet-fork/terraform/main.tf

```diff
@@ -0,0 +1,307 @@
+  }
+
+  lifecycle_policy {
+    transition_to_ia = "AFTER_30_DAYS"
+  }
+}
+
+resource "aws_efs_mount_target" "aztec_fork_private_az1" {
+  file_system_id  = aws_efs_file_system.aztec_mainnet_fork_data_store.id
+  subnet_id       = data.terraform_remote_state.setup_iac.outputs.subnet_az1_private_id
+  security_groups = [data.terraform_remote_state.setup_iac.outputs.security_group_private_id]
+}
+
+resource "aws_efs_mount_target" "aztec_fork_private_az2" {
+  file_system_id  = aws_efs_file_system.aztec_mainnet_fork_data_store.id
+  subnet_id       = data.terraform_remote_state.setup_iac.outputs.subnet_az2_private_id
+  security_groups = [data.terraform_remote_state.setup_iac.outputs.security_group_private_id]
+}
+
+# Define deployment task and service
+resource "aws_ecs_task_definition" "aztec_mainnet_fork" {
+  family                   = "aztec-network-mainnet-fork"
+  requires_compatibilities = ["FARGATE"]
+  network_mode             = "awsvpc"
+  cpu                      = "2048"
+  memory                   = "4096"
+  execution_role_arn       = data.terraform_remote_state.setup_iac.outputs.ecs_task_execution_role_arn
+
+  volume {
+    name = "efs-data-store"
+    efs_volume_configuration {
+      file_system_id = aws_efs_file_system.aztec_mainnet_fork_data_store.id
+    }
+  }
+
+  container_definitions = <<DEFINITION
+[
+  {
+    "name": "aztec-network-mainnet-fork",
+    "image": "278380418400.dkr.ecr.eu-west-2.amazonaws.com/aztec-mainnet-fork:aztec3-packages-prod",
+    "essential": true,
+    "environment": [
+      {
+        "name": "API_KEY",
+        "value": "${var.FORK_API_KEY}"
+      },
+      {
+        "name": "MNEMONIC",
+        "value": "${var.FORK_MNEMONIC}"
+      },
+      {
+        "name": "INFURA_API_KEY",
+        "value": "${var.INFURA_API_KEY}"
+      },
+      {
+        "name": "CHAIN_ID",
+        "value": "31337"
+      },
+      {
+        "name": "SNAPSHOT_FREQUENCY",
+        "value": "15"
+      }
+    ],
+    "mountPoints": [
+      {
+        "containerPath": "/data",
+        "sourceVolume": "efs-data-store"
+      }
+    ],
+    "logConfiguration": {
+      "logDriver": "awslogs",
+      "options": {
+        "awslogs-group": "${aws_cloudwatch_log_group.aztec_mainnet_fork_logs.name}",
+        "awslogs-region": "eu-west-2",
+        "awslogs-stream-prefix": "ecs"
+      }
+    },
+    "portMappings": [
+      {
+        "containerPort": 80,
+        "hostPort": 80
+      }
+    ]
+  }
+]
+DEFINITION
+}
+
+
+# ALB to to limit public requests to apikey routes
+resource "aws_alb_target_group" "mainnet_fork" {
+  name                 = "aztec-network-mainnet-fork"
+  port                 = "80"
+  protocol             = "HTTP"
+  target_type          = "ip"
+  vpc_id               = data.terraform_remote_state.setup_iac.outputs.vpc_id
+  deregistration_delay = 5
+  depends_on = [
+    data.aws_alb.aztec-network_alb
+  ]
+
+  health_check {
+    path                = "/"
+    matcher             = "404,400"
```

Comment:
Shouldn't ngnix return a 200?

Reply:
So it should on the "/" path but that's actually the mistake, this should be `"/${var.FORK_API_KEY}"` to check that the fork is actually running, not just the container. The request will be forwarded to ANVIL which should return a 400 for GET request